### ISSUE TITLE
Allow CSS Variables theme

### ISSUE THREAD

**Issue by PuruVJ** [antfu/shikiji#67]
### Clear and concise description of the problem

Thanks for this library ❤️‍🔥

We're looking to replace shiki and shiki-twoslash with shikiji and shikiji-twoslash in `svelte.dev`, `kit.svelte.dev` and `learn.svelte.dev` sites. There, we use CSS variables heavily for syntax highlighting and theme switching.

```css
--sk-code-ts-bg: hsl(var(--sk-back-2-hsl));
--sk-code-base: hsl(45, 7%, 75%);
--sk-code-comment: hsl(0, 0%, 55%);
--sk-code-keyword: hsl(204, 88%, 65%);
--sk-code-function: hsl(19, 67%, 75%);
--sk-code-string: hsl(41, 37%, 68%);
--sk-code-number: hsl(120, 100%, 25%);
--sk-code-template-string: hsl(2, 80%, 47%);
--sk-code-tags: var(--sk-code-function);
--sk-code-important: var(--sk-code-string);
--sk-code-diff-base: hsla(0, 0%, 100%, 0.5);
--sk-code-diff-inserted: hsl(120, 46%, 48%);
--sk-code-diff-removed: hsl(2, 64%, 70%);
```

And the `--shiki-color-` variables are set accordingly:

```css
body,
body.light,
body.dark {
  --shiki-color-text: var(--sk-code-base);
  --shiki-color-background: hsl(var(--sk-back-3-hsl));
  --shiki-token-constant: var(--sk-code-base);
  --shiki-token-string: var(--sk-code-string);
  --shiki-token-comment: var(--sk-code-comment);
  --shiki-token-keyword: var(--sk-code-keyword);
  --shiki-token-parameter: var(--sk-code-base);
  --shiki-token-function: var(--sk-code-function);
  --shiki-token-string-expression: var(--sk-code-string);
  --shiki-token-punctuation: var(--sk-code-base);
  --shiki-token-link: var(--sk-code-keyword);
}
```

### Suggested solution

It would be nice to have some way to have CSS variables work, as in the codebase, these same variables(--sk-code-) are used in the svelte REPL as well. A big refactor which turns it to dual theme approach will be quite hard to do at this scale.

### Alternative

Cant move to shikiji without this

### Additional context

(I did commented on an issue with same content, but that didn't get any response, hence opening a new one)

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by antfu** [comment]
Tried a more agnositc approach #68, happy to hear what you think :)

**Comment by PuruVJ** [comment]
Wow that's a real quick PR! 👑

> Can you elaborate more on why this makes it hard to migrate? Is you want to keep exactly the same styling as the current theme strictly?

It would require hard-coding the colors into shiki theme as well, both light and dark. Those colors would have to coexist with the CSS variables, leading to duplication.

> Tried a more agnositc approach https://github.com/antfu/shikiji/pull/68, happy to hear what you think :)

This one is nice!! I noticed you mentioned reinstating the CSS Variables approach. That would be ideal, but if it's too much tech debt, I'd still be happy using colorReplacements approach. An example of doing that in the docs would be appreciated!

Again, thanks for such a quick response!

**Comment by antfu** [comment]
https://shikiji.netlify.app/guide/theme-colors#css-variables-theme

Let me know if this unblocks your usage
